dashboard/client/src/components/ModalRunOnMyOF/ModalRunOnMyOF.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import {
+  Modal,
+  ModalBody,
+  ModalHeader,
+  FormFeedback,
+  Form,
+  Input,
+  Button,
+  Row,
+  Col,
+} from 'reactstrap';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCopy } from '@fortawesome/free-solid-svg-icons';
+
+class ModalRunOnMyOF extends Component {
+  state = {
+    valid: false,
+  };
+
+  constructor() {
+    super();
+
+    this.handleCopyClick = this.handleCopyClick.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  handleCopyClick(e) {
+    e.preventDefault();
+
+    this.input.select();
+
+    document.execCommand('copy');
+
+    this.setState({
+      valid: true,
+    }, () => {
+      setTimeout(() => {
+        this.setState({
+          valid: false,
+        })
+      }, 1500)
+    })
+  };
+
+  closeModal() {
+    this.setState({ active: false });
+  }
+
+  render() {
+    const { shortName, image, gitOwner } = this.props.fn;
+    const code = `{
```

Comment:
In a follow-up PR could you extract this into a new function? (perhaps add a unit test too?)